evm/src/main/java/org/hyperledger/besu/evm/precompile/KZGPointEvalPrecompiledContract.java

```diff
@@ -0,0 +1,154 @@
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import ethereum.ckzg4844.CKZG4844JNI;
+import org.apache.tuweni.bytes.Bytes;
+import org.jetbrains.annotations.NotNull;
+
+public class KZGPointEvalPrecompiledContract implements PrecompiledContract {
+
+  public KZGPointEvalPrecompiledContract(final Optional<String> pathToTrustedSetup) {
+
+    String pathToSetup;
+    CKZG4844JNI.Preset bitLength;
+    if (pathToTrustedSetup.isPresent()) {
+      pathToSetup = pathToTrustedSetup.get();
+    } else {
+      InputStream is =
+          KZGPointEvalPrecompiledContract.class.getResourceAsStream(
+              "mainnet_kzg_trusted_setup_4096.txt");
+      try {
+        File jniWillLoadFrom = File.createTempFile("kzgTrustedSetup", "txt");
+        Files.copy(is, jniWillLoadFrom.toPath(), REPLACE_EXISTING);
+        is.close();
+        pathToSetup = jniWillLoadFrom.getAbsolutePath();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    try {
+      BufferedReader setupFile =
+          Files.newBufferedReader(Paths.get(pathToSetup), Charset.defaultCharset());
+      String firstLine = setupFile.readLine();
+      if ("4".equals(firstLine)) {
+        bitLength = CKZG4844JNI.Preset.MINIMAL;
+      } else if ("4096".equals(firstLine)) {
+        bitLength = CKZG4844JNI.Preset.MAINNET;
+      } else {
+        throw new IllegalArgumentException("provided file not a setup for either 4 or 4096 bits");
+      }
+      CKZG4844JNI.loadNativeLibrary(bitLength);
+      try {
+        CKZG4844JNI.loadTrustedSetup(pathToSetup);
+      } catch (RuntimeException alreadyLoaded) {
+        if (alreadyLoaded.getMessage().contains("Trusted Setup is already loaded")) {
+        } else {
+          throw alreadyLoaded;
+        }
+      }
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException(e);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @VisibleForTesting
+  public void tearDown() {
+    CKZG4844JNI.freeTrustedSetup();
+  }
+
+  @Override
+  public String getName() {
+    return "KZGPointEval";
+  }
+
+  @Override
+  public long gasRequirement(final Bytes input) {
+    // As defined in EIP-4844
+    return 50000;
+  }
+
+  @NotNull
+  @Override
+  public PrecompileContractResult computePrecompile(
+      final Bytes input, @NotNull final MessageFrame messageFrame) {
+
+    if (input.size() != 192) {
+      return new PrecompileContractResult(
+          Bytes.EMPTY,
+          false,
+          MessageFrame.State.COMPLETED_FAILED,
+          Optional.of(ExceptionalHaltReason.PRECOMPILE_ERROR));
+    }
+    // Bytes versionedHash = input.slice(0, 32);
```

Comment:
this should be used to assert the below part, right? Will it be implemented in another PR?
> 
>     # Verify commitment matches versioned_hash
>     assert kzg_to_versioned_hash(commitment) == versioned_hash

Reply:
bump